packages/components/src/combobox-control/index.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import { useCombobox } from 'downshift';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Dashicon } from '../';
+
+const itemToString = ( item ) => item && item.name;
+// This is needed so that in Windows, where
+// the menu does not necessarily open on
+// key up/down, you can still switch between
+// options with the menu closed.
+const stateReducer = (
+	{ selectedItem },
+	{ type, changes, props: { items } }
+) => {
+	switch ( type ) {
+		case useCombobox.stateChangeTypes.ToggleButtonKeyDownArrowDown:
+			// If we already have a selected item, try to select the next one,
+			// without circular navigation. Otherwise, select the first item.
+			return {
+				selectedItem:
+					items[
+						selectedItem ?
+							Math.min( items.indexOf( selectedItem ) + 1, items.length - 1 ) :
+							0
+					],
+			};
+		case useCombobox.stateChangeTypes.ToggleButtonKeyDownArrowUp:
+			// If we already have a selected item, try to select the previous one,
+			// without circular navigation. Otherwise, select the last item.
+			return {
+				selectedItem:
+					items[
+						selectedItem ?
+							Math.max( items.indexOf( selectedItem ) - 1, 0 ) :
+							items.length - 1
+					],
+			};
+		default:
+			return changes;
+	}
+};
+export default function ComboboxControl( {
+	className,
+	hideLabelFromVision,
+	label,
+	options: items,
+	onInputValueChange: onInputValueChange,
+	onChange: onSelectedItemChange,
+	value: _selectedItem,
+} ) {
+	const {
+		getLabelProps,
+		getToggleButtonProps,
+		getComboboxProps,
+		getInputProps,
+		getMenuProps,
+		getItemProps,
+		isOpen,
+		highlightedIndex,
+		selectedItem,
+	} = useCombobox( {
+		initialSelectedItem: items[ 0 ],
+		items,
+		itemToString,
+		onInputValueChange,
+		onSelectedItemChange,
+		selectedItem: _selectedItem,
+		stateReducer,
+	} );
+	const menuProps = getMenuProps( {
+		className: 'components-combobox-control__menu',
+	} );
+	// We need this here, because the null active descendant is not
+	// fully ARIA compliant.
+	if (
+		menuProps[ 'aria-activedescendant' ] &&
+		menuProps[ 'aria-activedescendant' ].slice( 0, 'downshift-null'.length ) ===
+			'downshift-null'
+	) {
+		delete menuProps[ 'aria-activedescendant' ];
+	}
+	return (
+		<div className={ classnames( 'components-combobox-control', className ) }>
+			{ /* eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */ }
+			<label
+				{ ...getLabelProps( {
+					className: classnames( 'components-combobox-control__label', {
+						'screen-reader-text': hideLabelFromVision,
+					} ),
+				} ) }
+			>
+				{ label }
+			</label>
+			<div
+				{ ...getComboboxProps( {
+					className: 'components-combobox-control__button',
+				} ) }
+			>
+				<input
+					{ ...getInputProps( {
+						className: 'components-combobox-control__button-input',
+					} ) }
+				/>
+				<Button
```

Comment:
Again, if we're following [Example 1](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html), I don't think we need a button here at all.

Reply:
I think it's valuable to have a button on the right to open the list without typing, isn't it?

It's what Downshift recommends too.

cc @silviuavram

Reply:
I'm going by the WAI-ARIA patterns linked to above, where the only example with a button is the inline autocomplete one. That's not the one we're implementing here, so thought it would be good to stay consistent with their recommendations.

Reply:
Yes, there are versions with the toggle and versions without. The toggle is helpful to see the options before typing anything, but if that is not part of the requirement then it can be ommited. :)

Reply:
Should we keep it? It looks quite nice.

Reply:
Let's keep it for now, shouldn't do any harm 🙂